### DISCUSSION TITLE
Update Sortable Table when TABLE with same name gets replace with new structure & data

### DISCUSSION THREAD

**Discussion by Robert-M-Muench** [uwdata/mosaic#291] [Q&A]
I'm using https://uwdata.github.io/mosaic/examples/table.html on a single-page app, which resets MOSAIC's state with:
```
vg.coordinator().clear();
executeQuery("DROP TABLE mydata")
```
so that I can load new data (via drag-and-drop) without having to reload the page. This works for plotting a graph, but rendering the data table once again uses the previous SQL query, which, of course, fails because `mydata` now looks differently.

How can I tell the table renderer to generate a new SQL statement?

**Comment by jheer** [comment]
To make sure I'm understanding what is going on, I assume you are also regenerating all the Mosaic components? If so, and you are regenerating the `table(...)` component, it should not have access to any prior SQL queries directly.

However, if the schema of your data table is changing, you might be getting stale data from the Coordinator's table catalog cache. When you call `clear()` on the coordinator with no argument, all of the existing clients (including tables and plots) should get disconnected and the query cache is cleared. However, as the definition of tables is not expected to change (in the default case), the coordinator still hangs on to previously loaded table schema information.

So, maybe try this instead?

```js
vg.coordinator().clear({ clients: true, cache: true, catalog: true });
```

This will force the table catalog to also be cleared, causing the database to be queried anew when looking up table column information.
